### ISSUE TITLE
Noticed a breaking change when enabling/disabling actions for certain resources

### ISSUE THREAD

**Issue by paulandroshchuk** [laravel/nova-issues#6105]
- Nova Version: 4.32.9

### Description:
At 4.32.8 you guys fixed this issue ([see changelog](https://nova.laravel.com/releases/4.32.8)):
> Fixed an issue where actions that could not be run were still showing in the action dropdown.

Previously, when we had `canRun()` return `false`, the action would not show up in the dropdown:
```
(new SendWelcomeEmail)
    ->onlyOnDetail()
   // Before 4.32.9 it would not show up in the dropdown
    ->canRun(fn () => false),
```

Now, after that fix, it shows that action in the dropdown (but it's disabled now):
![CleanShot 2023-12-13 at 21 53 25@2x](https://github.com/laravel/nova-issues/assets/13240539/707476e1-5bf3-4965-8dd1-c2d45d774c6a)

Is it really intended now that these actions will always show up now when they can't be run? Even though they're disabled now.

This change introduced a bug in our system as we had some custom logic inside the `{AnyNovaAction}@fields` method. The thing is before that update, Nova would not call that method if the action cannot be run. Now, it will always run that method even though that action is disabled now.

### Steps to reproduce
1. Downgrade to 4.32.7
2. Create an action
3. Add custom logic to the action's `fields()` method (put some logging just for example)
4. Hardcode the `canRun()` to always return `false`
5. Open that Nova resource and you won't see that action in the dropdown and the `fields()` method won't run.
6. Upgrade to latest version
7. Open the same Nova resource and you'll see that now it shows the action (disabled) and runs that method `fields()`.

**Comment by crynobone** [comment]
> Is it really intended now that these actions will always show up now when they can't be run? Even though they're disabled now.

Yes, see https://github.com/laravel/nova-issues/discussions/5157

**Comment by crynobone** [comment]
We could look into just returning an empty array instead of resolving `fields` if `Action::$authorizedToRunAction` is `false`.

**Comment by hivokas** [comment]
> We could look into just returning an empty array instead of resolving `fields` if `Action::$authorizedToRunAction` is `false`.

That would be a good solution I think.

**Comment by hivokas** [comment]
@crynobone we have `Account` model. If account's `is_subaccount` field is `true`, then all 25 actions are available. If `is_subaccount` field is `false`, then only 5 actions are available. With these new updates we can't use `canSee` to hide these 20 unavailable actions because `canSee` doesn't receive `$model`. If we use `canRun`, it will keep showing these 20 actions but in a disabled state, which feels unnecessary.

As a compromise, what do you think about such behavior?

- if `canSee` is not set, `canRun` returns false, then we'll hide the action (useful for situation described above);
- if `canSee` returns false, `canRun` returns false, then we'll hide the action;
- if `canSee` returns true, `canRun` returns false, then we'll show the action in disabled state (useful for https://github.com/laravel/nova-issues/discussions/5157#discussion-4655857).

**Comment by paulandroshchuk** [comment]
Or could we have it as a global setting? Just something like this `Nova::hideWhenTheyCantRun()`.

**Comment by davidhemphill** [comment]
Can you update your actions to use `canSee` since that's what it's used for?

**Comment by paulandroshchuk** [comment]
It would be a fine workaround if the resource was passed to the `canSee` callback, [according to one of the comments above](https://github.com/laravel/nova-issues/issues/6105#issuecomment-1855604647).
